src/sage/rings/number_field/order.py

```diff
@@ -119,6 +119,83 @@ def quadratic_order_class_number(disc):
         h = pari.qfbclassno(disc)
     return ZZ(h)
 
+def quadratic_order_approximate_class_number(disc, *, bound=10**4):
+    r"""
+    Return *an approximation of* the class number of
+    the quadratic order of given discriminant.
+
+    Currently only implemented for maximal orders
+    in imaginary-quadratic fields.
+
+    EXAMPLES::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: QuadraticField(-419).class_number()
+        9
+        sage: quadratic_order_approximate_class_number(-419)
+        9.0...
+
+    ::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: d = 100000000000031
+        sage: QuadraticField(-d).class_number(proof=False)
+        14414435
+        sage: round(quadratic_order_approximate_class_number(-d))
+        144...
+        sage: round(quadratic_order_approximate_class_number(-d, bound=10**6))
+        1441...
+
+    ::
+
+        sage: from sage.rings.number_field.order import quadratic_order_approximate_class_number
+        sage: # Test it against the exact class number computed for the CSIDH-512 prime
+        sage: # Source: https://eprint.iacr.org/2019/498.pdf
+        sage: p = 4 * prod(primes(3,374)) * 587 - 1
+        sage: hreal = 84884147409828091725676728670213067387206838101828807864190286991865870575397
+        sage: assert not hreal * BQFClassGroup(-p).random_element()
+        sage: h = round(quadratic_order_approximate_class_number(-p, bound=10**3)); h
+        8...
+        sage: RR(h / hreal)
+        1.00...
+        sage: h = round(quadratic_order_approximate_class_number(-p, bound=10**4)); h
+        84...
+        sage: RR(h / hreal)
+        0.99...
+        sage: h = round(quadratic_order_approximate_class_number(-p, bound=10**5)); h
+        848...
+        sage: RR(h / hreal)
+        0.999...
```

Comment:
at least the formula should be deterministic so there shouldn't be a flaky test issue here, but for your information there's `# abs tol 1e-3`.